transport/http/jsonrpc/request_response_types.go

```diff
@@ -35,6 +35,16 @@ func (id *RequestID) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+func (id *RequestID) MarshalJSON() ([]byte, error) {
+	if id.intError == nil {
+		return json.Marshal(id.intValue)
+	} else if id.floatError == nil {
+		return json.Marshal(id.floatValue)
+	} else {
+		return json.Marshal(id.stringValue)
+	}
+}
```

Comment:
My sense is this would be better expressed without the short-circuit logic, as

```go
switch {
case id.intError == nil && id.floatError != nil:
    return json.Marshal(id.intValue)
case id.intError != nil && id.floatError == nil:
    return json.Marshal(id.floatValue)
default:
    return json.Marshal(id.stringValue)
}
```

What do you think?

Reply:
I dont think so, intError and floatError can both be nil

Reply:
Yes, if intErr == nil and floatErr == nil, the final default case is triggered.

Reply:
For integer value, intErr and floatErr are nil and the default case will return empty string.

Reply:
Ah, OK, I understand. Then the original implementation is best.

Reply:
Awesome!